shiny/render/renderer/_renderer.py

```diff
@@ -404,3 +408,19 @@ def get_sync_fn(self) -> Callable[[], IT | None]:
             )
         sync_fn = cast(Callable[[], IT], self._orig_fn)
         return sync_fn
+
+    @contextmanager
+    def _current_output_id(self):
+        from ...session import get_current_session
+
+        session = get_current_session()
+        if session is None:
+            yield
+            return
+
+        old_id = session._current_output_id
+        try:
+            session._current_output_id = self._renderer.output_id
+            yield
+        finally:
+            session._current_output_id = old_id
```

Comment:
I like this approach!, but I feel it is in the wrong place.

👀 for a different location.

Reply:
IIRC, we ended up not finding a better place?